openedx/features/discounts/tests/test_discount_restriction_models.py

```diff
@@ -0,0 +1,103 @@
+from student.tests.factories import UserFactory
+
+
+@ddt.ddt
+class TestDiscountRestrictionConfig(CacheIsolationTestCase):
+    """
+    Test discount restriction config
+    """
+    ENABLED_CACHES = ['default']
+
+    def setUp(self):
+        self.course_overview = CourseOverviewFactory.create()
+        self.user = UserFactory.create()
+        super(TestDiscountRestrictionConfig, self).setUp()
+
+    @ddt.data(True, False)
+    def test_enabled_for_course(
+        self,
+        enabled,
+    ):
+        DiscountRestrictionConfig.objects.create(
+            enabled=enabled,
+            course=self.course_overview,
+        )
+        course_key = self.course_overview.id
+
+        self.assertEqual(
+            enabled,
+            DiscountRestrictionConfig.current(course_key=course_key).enabled
+        )
+
+    @ddt.data(
+        # Generate all combinations of setting each configuration level to True/False/None
+        *itertools.product(*[(True, False, None)] * 4)
+    )
+    @ddt.unpack
+    def test_config_overrides(self, global_setting, site_setting, org_setting, course_setting):
+        """
+        Test that the stacked configuration overrides happen in the correct order and priority.
+
+        This is tested by exhaustively setting each combination of contexts, and validating that only
+        the lowest level context that is set to not-None is applied.
+        """
+        # Add a bunch of configuration outside the contexts that are being tested, to make sure
+        # there are no leaks of configuration across contexts
+        non_test_course_enabled = CourseOverviewFactory.create(org='non-test-org-enabled')
+        non_test_course_disabled = CourseOverviewFactory.create(org='non-test-org-disabled')
+        non_test_site_cfg_enabled = SiteConfigurationFactory.create(
+            values={'course_org_filter': non_test_course_enabled.org}
+        )
+        non_test_site_cfg_disabled = SiteConfigurationFactory.create(
+            values={'course_org_filter': non_test_course_disabled.org}
+        )
+
+        DiscountRestrictionConfig.objects.create(course=non_test_course_enabled, enabled=True)
+        DiscountRestrictionConfig.objects.create(course=non_test_course_disabled, enabled=False)
+        DiscountRestrictionConfig.objects.create(org=non_test_course_enabled.org, enabled=True)
+        DiscountRestrictionConfig.objects.create(org=non_test_course_disabled.org, enabled=False)
+        DiscountRestrictionConfig.objects.create(site=non_test_site_cfg_enabled.site, enabled=True)
+        DiscountRestrictionConfig.objects.create(site=non_test_site_cfg_disabled.site, enabled=False)
+
+        # Set up test objects
+        test_course = CourseOverviewFactory.create(org='test-org')
+        test_site_cfg = SiteConfigurationFactory.create(values={'course_org_filter': test_course.org})
+
+        DiscountRestrictionConfig.objects.create(enabled=global_setting)
```

Comment:
We talked about removing this global setting and changing it to always enabled.

Reply:
i believe we said we would not need to add enable things on a site level for edx.org  
the site field is part of stackedconfigmodels so if we want to use stackedconfigmodels we should make sure all the fields are working properly here, even if edx.org won't use the site field.  
the test above this one though does not create a config for site